### ISSUE TITLE
Docker app improvements

### ISSUE THREAD

**Issue by Schluggi** [truenas/apps#904]
Here are my list of improvements I would like to see in the future:

1. Add the option to add custom images for custom / yaml based apps. To replace the default images 
![image](https://github.com/user-attachments/assets/77f7aaaa-f2b3-461e-bc70-462b41512eb3) ![image](https://github.com/user-attachments/assets/650fe13c-38ce-4dae-b94a-dcae5f4abd28)

2. Add an option to full-screen the docker-compose files while editing. It does not make any fun at the moment because of the small width. 
![image](https://github.com/user-attachments/assets/05a67c6e-ce10-4d50-913e-6b1572b5cff5)

3. Auto-Image updates. Add an option to automatically update apps. Or is there and cli command for that I can use?

4. Make contributing more newbie friendly. At least for now I was not able to spot an official "how to release a new app"-guide.

5. Make pg versions selectable. I was unable to migrate all my apps, because some postgres version mismatches. Would be nice to have the option between some major versions of an app.
For some apps like Nextcloud, mariadb is the [recommend](https://docs.nextcloud.com/server/latest/admin_manual/installation/system_requirements.html) database engine. This should be also selectable.
![image](https://github.com/user-attachments/assets/a5a8b5f8-62ca-4483-aab4-43868ebf4f02)

**Comment by stavros-k** [comment]
> 4\. Make contributing more newbie friendly. At least for now I was not able to spot an official "how to release a new app"-guide.

As you may have noticed there is still a warning for contributions at the top of the readme file.
Once everything is ready, there will be some basic steps outlined.

> 5\. Make pg versions selectable. I was unable to migrate all my apps, because some postgres version mismatches. Would be nice to have the option between some major versions of an app.
>         For some apps like Nextcloud, mariadb is the [recommend](https://docs.nextcloud.com/server/latest/admin_manual/installation/system_requirements.html) database engine. This should be also selectable.


Multiple db engines per app will not be supported. If you want to choose another engine, you are free to use one of the custom options (portainer, dockge, custom app, etc). Also note that the "recommended" is towards the version IF you choose MySQL or MariaDB.

As for the versions, while it won't be selectable, at least there are no plans for that. But there are plans to upgrade existing installations. (New installations will use newer versions).

If we allow users to select versions, then upgrading them will become impossible, and users will have to manually update versions. Which I don't think that makes sense on prepackage apps.
Until then your option is custom app if you want another pg version _now_

---

For the first 3, I'd suggest you to ask on the forums (Under feature Requests). As it is not something that can be done on the apps side.

Thanks

**Comment by stavros-k** [comment]
I'll close this one, as there is already an issue for tracking the pg version upgrades. 
The rest are not something actionable

Thanks!
